securedrop_client/db.py

```diff
@@ -278,11 +329,21 @@ def __str__(self) -> str:
         if self.content is not None:
             return self.content
         else:
+            if self.download_error is not None:
+                return self.download_error.explain(self.__class__.__name__)
             return '<Reply not yet available>'
 
     def __repr__(self) -> str:
         return '<Reply {}: {}>'.format(self.uuid, self.filename)
 
+    def download_failed_since(self, interval: datetime.timedelta) -> bool:
+        """
+        Returns True if the reply has a download error and was updated since `interval` ago.
+        """
+        return (
+            self.download_error and self.last_updated > (datetime.datetime.utcnow() - interval)
```

Comment:
Gotcha. So we expect the last_updated field to be updated if there is an error during job execution, and we also expect that the database object for a failed Message, File, or Reply will not be updated at any other time except when a DownloadJob gets retried... I'll keep an eye out for this as I review to make sure this is True, but I believe it is.